server/libs/infra/src/entities/partner.entity.ts

```diff
@@ -0,0 +1,26 @@
+import { CreateDateColumn, Entity, ManyToOne, PrimaryColumn, JoinColumn, UpdateDateColumn } from 'typeorm';
+
+import { UserEntity } from './user.entity';
+
+@Entity('partners')
+export class PartnerEntity {
+  @PrimaryColumn('uuid')
+  sharedById!: string;
+
+  @PrimaryColumn('uuid')
+  sharedWithId!: string;
+
+  @ManyToOne(() => UserEntity, (user) => user.sharedBy, { onDelete: 'CASCADE' })
+  @JoinColumn({ name: 'sharedById' })
+  sharedBy!: UserEntity;
+
+  @ManyToOne(() => UserEntity, (user) => user.sharedWith, { onDelete: 'CASCADE' })
+  @JoinColumn({ name: 'sharedWithId' })
+  sharedWith!: UserEntity;
+
+  @CreateDateColumn({ type: 'timestamptz' })
+  createdAt?: string;
+
+  @UpdateDateColumn({ type: 'timestamptz' })
+  updatedAt?: string;
```

Comment:
Both these fields are *not* nullable in the database and that should be reflected here. I'm also not too sure if the `string` type is correct, from the [TypeORM docs](https://typeorm.io/decorator-reference#createdatecolumn) it looks to be `Date`

Reply:
Thank you! I see several other entities also use `string` type. Looks like some kind of tech-debt